File: Source/Fantasy.Jobs/Resources/ResourceHandle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;

namespace Fantasy.Jobs.Resources
{
    internal class ResourceHandle : IResourceHandle
    {
        public IResourceService ResourceService {get;set;}

        public bool SuspendEngine { get; set; }

        public ResourceParameter[] Parameters { get; set; }

        public Guid Id { get; set; }

        private bool _disposed = false;

        public void Dispose()
        {
            if(!_disposed)
            {
                _disposed = true;
                this.ResourceService.Release(this); 
            }
        }

        public event EventHandler<RevokeArgs> Revoke;

        public virtual void OnRevoke(RevokeArgs e)
        {
            if (this.Revoke != null)
            {
                this.Revoke(this, e);
            }
        }
        
    }
}

File: AutoConverted/Fantasy.Jobs/Resources/ResourceHandle.java
package Fantasy.Jobs.Resources;

public class ResourceHandle implements IResourceHandle
{
	private IResourceService privateResourceService;
	public final IResourceService getResourceService()
	{
		return privateResourceService;
	}
	public final void setResourceService(IResourceService value)
	{
		privateResourceService = value;
	}

	private boolean privateSuspendEngine;
	public final boolean getSuspendEngine()
	{
		return privateSuspendEngine;
	}
	public final void setSuspendEngine(boolean value)
	{
		privateSuspendEngine = value;
	}

	private ResourceParameter[] privateParameters;
	public final ResourceParameter[] getParameters()
	{
		return privateParameters;
	}
	public final void setParameters(ResourceParameter[] value)
	{
		privateParameters = value;
	}

	private Guid privateId = new Guid();
	public final Guid getId()
	{
		return privateId;
	}
	public final void setId(Guid value)
	{
		privateId = value;
	}

	private boolean _disposed = false;

	public final void dispose()
	{
		if(!_disposed)
		{
			_disposed = true;
			this.getResourceService().Release(this);
		}
	}

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	public event EventHandler<RevokeArgs> Revoke;

	public void OnRevoke(RevokeArgs e)
	{
		if (this.Revoke != null)
		{
			this.Revoke(this, e);
		}
	}

}